### ISSUE TITLE
Add Response setting to System Settings

### ISSUE THREAD

**Issue by scloo** [ICTatRTI/coconut-analytics#193]
- Notification (48 hours)
- TBD
-Followup (72 hours)

**Comment by gmcressman** [comment]
This is related to the graphs in #143 and #144.

The current response protocol in Zanzibar has the following **targets**:

T1: Time from diagnosis to notification (entered into system) = 24 hours
T2: Time from notification received (by system) to follow-up completed = 48 hours

T1 is graphed in #143 using the following time thresholds: 

1. Within 24 hours (Target)
2. 25 to 72 hours
3. Over 72 hours
4. Not followed up

T2 is graphed in #144 using the following time thresholds:

1. Within 48 hours (Target)
2. 3 to 7 days
3. More than 1 week
4. Not followed up

The "1-3-7" system initiated in China is also being used in other countries in Asia and Southeast Asia. This sets three time targets as follows:

T1: **Case reporting within 1 day.** Any confirmed and suspected malaria cases by law must be reported to the web-based health information system within 24 hours of diagnosis by the local health-care provider.
T2: **Case investigation within 3 days.** All malaria cases should be confirmed and visited by the county-level China CDC, where the case is reported within three days, to determine where the case originated (local or imported).
T3: **Focus investigation and action within 7 days.** The focus investigation should be conducted as soon as possible. _If local transmission is possible or confirmed, targeted action to seek out other infections and reduce the chance of onward transmission is completed within seven days by the county-level China CDC of the county where the patient resides and/or works._

The 1-3-7 approach in China appears to have been helpful, at least for T1 and T2. They still struggle to meet the T3 7-day target. The 1-3-7 approach is being considered for use in Cambodia and other countries in the Asia-Pacific region, although the target response times will need to be adjusted for each context. 

T1 and T2 In Zanzibar correspond to T1 and T2 in the 1-3-7 approach. T3, the focus investigation and action, is triggered in Zanzibar by the following field in the household question set:

FollowupNeighbors [Yes|No]

That field should be set depending on the risk stratification of the area. In the case of Zanzibar, I believe risk is at the level of the District. If the index case household is in a high risk area, the current protocol in Zanzibar is to screen all households within 50 paces.

We also have the following field in the household question set:

Reasonforvisitinghousehold [Index Case Household | Mass Screen | ??]

In effect, T2 and T3 in Zanzibar are currently the same. That is, the include both the case investigation and, depending on risk level in the area of the household, the focus investigation.

It doesn't appear to me that we currently have any way in Coconut Surveillance to monitor compliance with T3, the focus investigation. We could look at the relationship between FollowupNeighbors with Reasonforvisitinghousehold. We could link households  and perhaps query all households in the database within a given radius of an index case household and a given time span. All that is possible, but may be more complicated than necessary. That said, we don't appear to have a method in Coconut Surveillance to measure compliance with T3, even if we have a T3.

In conclusion, the response time targets are very important. To support the existing protocol in Zanzibar and the 1-3-7 approach of China, I believe we need the following response target settings in the System Settings:

Response time targets

T1: Case reporting
T2: Case investigation
T3: Focus investigation

All three could be in set units of hours.

For each, we also need system settings for graphing and reporting, as described above. These might be:

1. t1 (Target, e.g. T1)
2. >t1 to <= t2
3. >t2
4. Not followed up

This is something else that we need to discuss with Mike and/or Jeremiah.

**Comment by gmcressman** [comment]
Some further information on 1-3-7:

https://idpjournal.biomedcentral.com/articles/10.1186/s40249-015-0089-2

> The China’s 1-3-7 strategy was initiated and extensively adopted in different types of counties (geographic regions) for reporting of malaria cases within 1 day, their confirmation and investigation within 3 days, and the appropriate public health response to prevent further transmission within 7 days. Assessing the level of compliance to the 1-3-7 strategy at the county level is a first step towards determining whether the surveillance and response strategy is happening according to plan. This study assessed if the time-bound targets of the 1-3-7 strategy were being sustained over time. Such information would be useful to improve implementation of the 1-3-7 strategy in China.
